fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanRepaymentScheduleProcessingWrapper.java

```diff
@@ -213,6 +213,10 @@ private Money cumulativePenaltyChargesDueWithin(final LocalDate periodStart, fin
                     cumulative = cumulative.plus(loanChargeAmt);
                 } else if (loanCharge.isDueForCollectionFromAndUpToAndIncluding(periodStart, periodEnd)) {
                     cumulative = cumulative.plus(loanCharge.amount());
+                    // Special case for Loan Charges (Due Date) added the same disbursement date
+                } else if (period.isFirstPeriod()
```

Comment:
Why don't we check directly whether the charge is due on disbursement date? This condition is kinda confusing to me and probably to you too, that's why you put the comment there.

Let's make it crystal clear, loan charge on disbursement date.

Reply:
This is a special case or requirement for the first installment, the other installments must to be in the other if

Reply:
i dont think this newly added branch will ever be executed.

the condition in line 214 will always be true before it could reach the new branch in line 217.
Also both of them do exactly the same thing. 
@josehernandezfintecheandomx could you advise on this?

Reply:
I would rather do this:
- If this is the first period
AND
- first period startdate == loanCharge due date

I dont think it is wrong your approach, but i feel not straightforward to do this `loanCharge.isDueForCollectionFromIncludingAndUpToAndIncluding` which is hard to understand what it is doing.

Also this special case is only when the charge due date is equals with first instalment start date. 
@josehernandezfintecheandomx what do you think?

Reply:
Well, It can be implemented in different ways. This one was used for the Fees in the previous change and now is the same code in the Penalties method. Plus the logic for validating the dates was reusing the same functions.

Based on that, I've updated the code as you suggest and I've removed the code added in the previous change that is not more needed. @adamsaghy